nvdaHelper/remote/excel.cpp

```diff
@@ -0,0 +1,472 @@
+#include <common/COMUtils.h>
+#include "inProcess.h"
+#include "nvdaInProcUtils.h"
+
+
+// Excel IDispatch IDs
+const long XLDISPID_RANGE_FORMULA=261;
+const long XLDISPID_RANGE_ITEM=170;
+const long XLDISPID_RANGE_ROW=257;
+const long XLDISPID_RANGE_COLUMN=240;
+const long XLDISPID_RANGE_ADDRESS=236;
+const long XLDISPID_RANGE_MERGEAREA=1385;
+const long XLDISPID_RANGE_NEXT=502;
+const long XLDISPID_RANGE_WIDTH=122;
+const long XLDISPID_RANGE_SHRINKTOFIT=209;
+const long XLDISPID_RANGE_WRAPTEXT=276;
+const long XLDISPID_RANGE_HYPERLINKS=1393;
+const long XLDISPID_HYPERLINKS_COUNT=118;
+const long XLDISPID_RANGE_ENTIREROW=247;
+const long XLDISPID_RANGE_ROWS=258;
+const long XLDISPID_ROWS_COUNT=118;
+const long XLDISPID_ROWS_ITEM=170;
+const long XLDISPID_ROW_SUMMARY=273;
+const long XLDISPID_ROW_SHOWDETAIL=585;
+const long XLDISPID_ROW_OUTLINELEVEL=271;
+const long XLDISPID_RANGE_ENTIRECOLUMN=246;
+const long XLDISPID_RANGE_COLUMNS=241;
+const long XLDISPID_COLUMNS_COUNT=118;
+const long XLDISPID_COLUMNS_ITEM=170;
+const long XLDISPID_COLUMN_SUMMARY=273;
+const long XLDISPID_COLUMN_SHOWDETAIL=585;
+const long XLDISPID_COLUMN_OUTLINELEVEL=271;
+const long XLDISPID_RANGE_COMMENT=910;
+const long XLDISPID_COMMENT_TEXT=138;
+const long XLDISPID_RANGE_HASFORMULA=267;
+const long XLDISPID_RANGE_VALIDATION=1387;
+const long XLDISPID_VALIDATION_TYPE=108;
+const long XLDISPID_VALIDATION_INPUTMESSAGE=1611;
+const long XLDISPID_VALIDATION_INPUTTITLE=1612;
+const long XLDISPID_RANGE_WORKSHEET=348;
+const long XLDISPID_WORKSHEET_PROTECTCONTENTS=292;
+const long XLDISPID_RANGE_LOCKED=269;
+const long XLDISPID_RANGE_TEXT=138;
+const long XLDISPID_RANGE_FONT=146;
+const long XLDISPID_FONT_SIZE=104;
+const long XLDISPID_FONT_BOLD=96;
+const long XLDISPID_FONT_ITALIC=101;
+const long XLDISPID_FONT_UNDERLINE=106;
+const long XLDISPID_FONT_STRIKETHROUGH=105;
+const long XLDISPID_FONT_NAME=110;
+const long XLDISPID_RANGE_DISPLAYFORMAT=666;
+const long XLDISPID_RANGE_OFFSET=254;
+
+const long NVCELLINFOFLAG_ADDRESS=0x1;
+const long NVCELLINFOFLAG_TEXT=0x2;
+const long NVCELLINFOFLAG_INPUTMESSAGE=0x4;
+const long NVCELLINFOFLAG_STATES=0x8;
+const long NVCELLINFOFLAG_COORDS=0x10;
+const long NVCELLINFOFLAG_OUTLINELEVEL=0x20;
+const long NVCELLINFOFLAG_COMMENTS=0x40;
+const long NVCELLINFOFLAG_FORMULA=0x80;
+const long NVCELLINFOFLAG_ALL=0xffff;
+
+// NVDA states
+const __int64 NVSTATE_EXPANDED=0x100;
+const __int64 NVSTATE_COLLAPSED=0x200;
+const __int64 NVSTATE_LINKED=0x1000;
+const __int64 NVSTATE_HASPOPUP=0x2000;
+const __int64 NVSTATE_PROTECTED=0x4000;
+const __int64 NVSTATE_HASFORMULA=0x1000000000;
+const __int64 NVSTATE_HASCOMMENT=0x2000000000;
+const __int64 NVSTATE_CROPPED=0x8000000000;
+const __int64 NVSTATE_OVERFLOWING=0x10000000000;
+const __int64 NVSTATE_UNLOCKED=0x20000000000;
+
+long getCellTextWidth(HWND hwnd, IDispatch* pDispatchRange) {
+	CComBSTR sText;
+	_com_dispatch_raw_propget(pDispatchRange,XLDISPID_RANGE_TEXT,VT_BSTR,&sText);
```

Comment:
Many (most, all?) are not checking the return result. Please add:
```suggestion
	auto res = _com_dispatch_raw_propget(pDispatchRange,XLDISPID_RANGE_TEXT,VT_BSTR,&sText);
	if( res !=S_OK){
		LOG_DEBUGWARNING(L"range text failed");
		return 0;
	}
```